doc/channel-plugin.md

```diff
@@ -0,0 +1,58 @@
+## Channel Plugin
+
+Channel plugins are processes that run independently of each other.
+
+Channel plugin uses stdin and stdout as json format to communicate. The request and response structure is inspired by
+JSON-RPC.
+
+### Request
+
+The request must be in json format and should contain following keys:
+
+- `method`: The request method to call
+- `params`: The params for request method
+- `id`: Unsigned int value. Require to assign the response to it's request as responses can be out of order
```

Comment:
**Required** to assign the response to **its** request as responses can be **sent** out of order